core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/ReactorFlowManager.java

```diff
@@ -454,6 +461,17 @@ public void triggerDetermineDatalakeDataSizes(Long stackId, String operationId)
         reactorNotifier.notify(stackId, selector, new DetermineDatalakeDataSizesBaseEvent(selector, stackId, operationId));
     }
 
+    public FlowIdentifier triggerSecretRotation(String crn, List<SecretType> secretTypes, RotationFlowExecutionType executionType) {
+        String selector = EventSelectorUtil.selector(SecretRotationFlowChainTriggerEvent.class);
+        Optional<ResourceBasicView> resourceBasicViewByResourceCrn = stackService.getResourceBasicViewByResourceCrn(crn);
```

Comment:
I think this part belongs to StackOperationService instead and this method should get the stackId as a parameter.